internal/utils/nodeGroup.go

```diff
@@ -15,6 +18,9 @@ import (
 	danav1alpha1 "nodeQuotaSync/api/v1alpha1"
 )
 
+// CalculateNodeGroup calculates the resource list for a node group based on the provided nodes, NodeQuotaConfig, and node group name.
+// It takes a context, a NodeList containing the nodes, the NodeQuotaConfig, and the node group name.
+// It returns the calculated resource list (v1.ResourceList) for the node group.
 func CalculateNodeGroup(ctx context.Context, nodes v1.NodeList, config danav1alpha1.NodeQuotaConfig, nodeGroup string) v1.ResourceList {
 	var ResourceMultiplier map[string]string
```

Comment:
Would probably be better to move this to a different function with a single responsibility that just returns the ResourceMultiplier